vector/record.go

```diff
@@ -0,0 +1,55 @@
+package vector
+
+import (
+	"github.com/brimdata/zed"
+	"github.com/brimdata/zed/zcode"
+)
+
+// XXX need to create memory model
+type mem struct{}
```

Comment:
What is this for?

Reply:
Looks like it's just a way to add no-op Ref and Unref methods to the types that embed this.

Reply:
Are the Ref/Unref methods some existing idiom in the codebase?

Reply:
Yes, they appear in zbuf.Batch.

https://github.com/brimdata/zed/blob/40639dce6c333da1c55844b026052dc3718cc2d2/zbuf/batch.go#L12-L34